packages/ts/react-crud/test/autoform.spec.tsx

```diff
@@ -408,194 +401,43 @@ describe('@hilla/react-crud', () => {
+          return <VerticalLayout>{children}</VerticalLayout>;
+        }
 
-    it('customLayoutRenderer is defined by FieldColSpan[][] and responsiveSteps, number of columns is based on responsiveSteps and colspan is based on each FieldColSpan', async () => {
-      const form = await populatePersonForm(
-        1,
-        {
-          responsiveSteps: [
-            { minWidth: '0', columns: 1 },
-            { minWidth: '800px', columns: 2 },
-            { minWidth: '1200px', columns: 3 },
-          ],
-          template: [
-            [
-              { property: 'firstName', colSpan: 1 },
-              { property: 'lastName', colSpan: 1 },
-              { property: 'email', colSpan: 1 },
-            ],
-            [
-              { property: 'someInteger', colSpan: 2 },
-              { property: 'someDecimal', colSpan: 1 },
-            ],
-          ],
-        },
-        'screen-1440-900',
-      );
-      expect(form.formLayout.responsiveSteps).to.have.length(3);
-      expect(form.formLayout.responsiveSteps).to.be.deep.equal([
-        { minWidth: '0', columns: 1 },
-        { minWidth: '800px', columns: 2 },
-        { minWidth: '1200px', columns: 3 },
-      ]);
-      await expectFieldColSpan(form, 'First name', '1');
-      await expectFieldColSpan(form, 'Last name', '1');
-      await expectFieldColSpan(form, 'Email', '1');
-      await expectFieldColSpan(form, 'Some integer', '2');
-      await expectFieldColSpan(form, 'Some decimal', '1');
+        const form = await populatePersonForm(1, { layoutRenderer: MyLayoutRenderer }, 'screen-1440-900');
+        expect(form.formLayout).to.not.exist;
+        const layout = await waitFor(() => form.renderResult.querySelector('vaadin-vertical-layout')!);
+        expect(layout).to.exist;
+      });
     });
 
-    it('customLayoutRenderer is defined as a function that renders the fields in a vertical manner, VerticalLayout is used instead of FormLayout', async () => {
-      function MyCustomLayoutRenderer({ children, form }: AutoFormLayoutRendererProps<PersonModel>) {
-        return <VerticalLayout>{children}</VerticalLayout>;
-      }
-
-      const form = await populatePersonForm(1, MyCustomLayoutRenderer, 'screen-1440-900');
-      expect(form.formLayout).to.not.exist;
-      const layout = await waitFor(() => form.renderResult.querySelector('vaadin-vertical-layout')!);
-      expect(layout).to.exist;
+    describe('visibleFields', () => {
+      it('renders the form according to visibleFields if defined', async () => {
+        const form = await populatePersonForm(1, { visibleFields: ['firstName', 'lastName', 'id', 'dummy'] });
+        const fields = await form.getFields('First name', 'Last name', 'Id');
+        const tagNames = fields.map((field) => field.localName);
+        expect(tagNames).to.eql(['vaadin-text-field', 'vaadin-text-field', 'vaadin-number-field']);
+        expect(form.queryField('Dummy')).to.be.undefined;
```

Comment:
Could also verify that some of the other properties are not rendered. Currently it just asserts that:
- There are fields for the three existing properties specified in `visibleFields`
- There is no field for the non-existing property

But it does not verify that no fields have been generated for the other existing properties that have not been specified in `visibleFields`.

Reply:
Good point